### PR TITLE
See #7593. Use connection_max to stop connections in rabbitmq

### LINKED ISSUE #7593: More predictable and firm connection_max for limiting number of connections per RabbitMQ node
Today it is possible to limit maximum allowed connections per vhost, per user, or per node.

Running rabbitMQ as a service, having limits like this is very useful, to avoid brokers running out of memory. The current back pressure mechanism for high memory usage (memory alarm) does not stop clients from creating new connections, and thus have no real effect at improving memory usage. 

The per vhost and user logic is handled by rabbitmq, and is set via runtime configuration, and
works as expected, with a nice error when you hit the limit.

For node level connection limit, rabbitmq passes along the limit value, set as config, to ranch. The config parameter is 'connection_max', and its existence is not really advertised.

This works, but has a few drawbacks.
* It is a 'soft' limit, basically max_connections + the number of acceptors.
* Hitting the limit will result in ranch not accepting anymore connections. This will not result in further connections being rejected, and thus no with a user friendly error message, as the kernel option allows queueing incoming connections. Instead, the attempt to create a connection will hang until a timeout is reached.

Suggestion: Similar to how rabbitmq can be configured to limit connections per vhost or per user, add the option to also
limit number of connections per node, handled within rabbitmq, to give a nice informative reason to the client of why 
a connection request is not accepted.

### PR DESCRIPTION
**Pull request by SimonUnge**
## Proposed Changes
Closes #7593 

Use connection_max to limit connections per node, from rabbitmq code, instead of ranch.

Add ranch_connection_max setting for setting ranch connection max.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [x] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

## Further Comments

N/A

### CODE REVIEW

File: deps/rabbit/src/rabbit_reader.erl [hunk 1]
```diff
@@ -1318,6 +1319,20 @@ is_vhost_alive(VHostPath, User) ->
                             [VHostPath, User#user.username, VHostPath])
     end.
 
+is_over_node_connection_limit(Port) ->
+    {Addr, _, _} = hd(rabbit_networking:tcp_listener_addresses(Port)),
+    Ref = rabbit_networking:ranch_ref(Addr, Port),
+    #{active_connections := ActiveConns} = ranch:info(Ref),
+    Limit = rabbit_misc:get_env(rabbit, connection_max, infinity),
```
Comment (lhoguin) [inline deps/rabbit/src/rabbit_reader.erl:1326]: `ranch_connection_max` no?

**Review by lhoguin — APPROVED** [review]
Nevermind the previous comment I understand now. Looks good to me, thanks!

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
We don't have a lot of details but in https://github.com/rabbitmq/rabbitmq-server/discussions/7787 this new function runs into an exception when a client connects.

**Comment by michaelklishin** [comment]
Here are [various config files](https://storage.bhs.cloud.ovh.net/v1/AUTH_dcaab5e32b234d56b626f72581e3644c/zuul_opendev_logs_7b7/878209/21/check/kolla-ansible-rocky9/7b7d31f/primary/logs/kolla_configs/rabbitmq/index.html)

**Comment by michaelklishin** [comment]
``` ini
RABBITMQ_NODENAME=rabbit@primary
RABBITMQ_LOG_BASE=/var/log/kolla/rabbitmq
RABBITMQ_DIST_PORT=25672
RABBITMQ_PID_FILE=/var/lib/rabbitmq/mnesia/rabbitmq.pid
RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-kernel inetrc '/etc/rabbitmq/erl_inetrc' +S 2:2 +sbwt none +sbwtdcpu none +sbwtdio none"
RABBITMQ_CTL_ERL_ARGS=""

export ERL_EPMD_ADDRESS=192.0.2.1
export ERL_EPMD_PORT=4369
export ERL_INETRC=/etc/rabbitmq/erl_inetrc
```

``` erl
[
  {kernel, [
    {inet_dist_use_interface, {192,0,2,1}},
    {inet_dist_listen_min, 25672},
    {inet_dist_listen_max, 25672}
  ]}
].
```

**Comment by michaelklishin** [comment]
A follow-up that will go into `3.11.13` https://github.com/rabbitmq/rabbitmq-server/pull/7788

**Pull request merged**